pandas/_libs/tslibs/nattype.pyx

```diff
@@ -151,9 +153,10 @@ cdef class _NaT(datetime):
             return self + neg_other
 
         elif getattr(other, '_typ', None) in ['period', 'series',
-                                              'periodindex', 'dateoffset']:
+                                              'periodindex', 'dateoffset',
+                                              'datetimearray',
```

Comment:
is this missing dti and tdi?

Reply:
is this true?

Reply:
I think so.  Same response as the other one: soon we're going to get rid of this list altogether (in fact, the implementation of `__sub__` will be identical to the implementation of `__add__`)